overlord/ifacestate/handlers.go

```diff
@@ -1532,6 +1534,121 @@ func (m *InterfaceManager) doHotplugDisconnect(task *state.Task, _ *tomb.Tomb) e
 	return nil
 }
 
+func (m *InterfaceManager) doHotplugAddSlot(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	systemSnap, err := systemSnapInfo(st)
+	if err != nil {
+		return fmt.Errorf("core snap not available")
+	}
+
+	ifaceName, hotplugKey, err := getHotplugAttrs(task)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot get hotplug task attributes: %s", err)
+	}
+
+	var slotSpec hotplug.RequestedSlotSpec
```

Comment:
this type here also need json encoding tags? it doesn't seem to have them atm

Reply:
Right, good idea to have explicit tags on them in case of future renames. Contrary to HotplugDeviceInfo, this struct has only exported fields, so it works, altough with automatically inferred json names.